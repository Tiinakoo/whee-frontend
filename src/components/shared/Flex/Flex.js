import React from "react";
import styles from "./Flex.module.scss";
import classNames from "classnames";

const Flex = ({
  className,
  children,
  centered,
  centeredVertically = centered,
  centeredHorizontally = centered,
  lineUpHorizontally,
  lineUpHorizontallyTo,
  ...props
}) => {
  const classNameParameters = [
    styles.flex,
    {
      [styles.flex__centeredVertically]: centeredVertically,
      [styles.flex__centeredHorizontally]: centeredHorizontally,
      [styles.flex__lineUpHorizontallyToMdScreen]:
        lineUpHorizontallyTo === "md",
      [styles.flex__lineUpHorizontally]: lineUpHorizontally,
    },
    className,
  ];

  return (
    <div className={classNames(classNameParameters)} {...props}>
      {children}
    </div>
  );
};

export default Flex;
