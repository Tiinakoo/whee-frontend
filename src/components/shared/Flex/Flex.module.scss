@import "../../../styles/mixins";
@import "../../../styles/baseline";

.flex {
  display: flex;
  flex-direction: column;

  &__centeredVertically {
    align-items: center;
  }

  &__centeredHorizontally {
    justify-content: center;
  }

  &__spaceBetween {
    justify-content: space-between;
  }

  &__lineUpHorizontallyToMdScreen {
    @include md {
      flex-direction: row;
      align-items: center;
    }
  }

  &__lineUpHorizontally {
    flex-direction: row;
  }

  &_flexItem {
    min-width: 1px;
    max-width: 100%;
    flex-grow: 1;
    flex-shrink: 1;
    flex-basis: 0%;
  }
}
