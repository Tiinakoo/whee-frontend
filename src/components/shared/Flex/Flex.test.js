import React from "react";

import { mount } from "enzyme";
import Flex from "./Flex";

describe("Flex", () => {
  it("renders", () => {
    const component = mount(<Flex />);

    expect(component).toMatchSnapshot();
  });
});
