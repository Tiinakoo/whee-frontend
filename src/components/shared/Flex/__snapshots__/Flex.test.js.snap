// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex renders 1`] = `
<Flex>
  <div
    className="flex"
  />
</Flex>
`;
